i18n/README.md

```diff
@@ -0,0 +1,37 @@
+# Translations (i18n)
+
+The UI Kit uses `react-intl` to manage the messages and translations. The core messages are defined in `messages.js` files, usually colocated with the related React component.
+
+The translations for the supported languages are then pulled into `./i18n/{en,de,es}.json` files.
+
+> **DO NOT EDIT THOSE FILES DIRECTLY, TRANSLATIONS SHOULD BE DONE ONLY IN TRANSIFEX!!!**
+
+### Translation software (Transifex)
+
+Translations should be managed using the Transifex software: https://www.transifex.com/commercetools/ui-kit/
+
+## Structure
+
+This folder contains the **JSON** files with the translated messages. The key of the messages corresponds to the `message.id` defined in each `messages.js`.
+
+#### Localized files
+
+There is one file for each language (e.g. `en`, `de`, `es`, etc.).
+
+#### Source file
+
+There is one special file `core.json`, which is used by Transifex as the **source file**. The messages in this file are used as the base for translating to the other locale. When you add new messages in the React components, those message keys will first go into the `core.json` (_read below_).
+
+## Adding new messages
+
+When you add new messages in the React components `messages.js`, you need to run `yarn i18n:build` which will extract the new messages and put them into the `core.json`.
+Then you can push and merge the changes to `master`. At this point, there is a step in CI that will push the source file `core.json` to Transifex thus making the new keys available to the system.
+
+## Translating messages
+
+When new messages are pushed to Transifex, you can start translating them using the [translation software](https://www.transifex.com/commercetools/ui-kit/).
+Once the messages have been translated and reviewed, you can pull them into the repository by running the command `yarn i18n:pull`. The command will sync the `{locale}.json` files in case something changed and will sync `en.json` -> `core.json`. This is to ensure that the english source messages are up-to-date when there needs to be a translation to another language.
```

Comment:
I see how this might be inconvenient for the ui-kit.

However, there is an easy fix to this: recently Tx added support for [translating branch specific versions](https://www.transifex.com/blog/2018/transifex-client-updates-branch-support-parallel-requests/), which means that we can push/pull from a branch.

Actually I've been meaning to try it out also in the MC. I guess we can give it a try first in the ui-kit. Let me update the scripts.

Reply:
So I did some testing and I don't like how it works.

If we want to continue using transifex, we would keep our current process and be aware that we would release components that might not have all translations.
If that's fine I'm ok with that.
If we absolutely want to have components fully translated all the time, we need to drop it then.

Reply:
@emmenko I ran into a "deadlock" while working on the date-inputs. I added tests to the date-inputs. The tests need to test the english and german locale. The english tests run fine, but the test for the german locale emit an `Missing message: "UIKit.DateInput.placeholder" for locale: "de", using default message as fallback.` warning. On CI they fail because of this warning. I would need to merge them to `master` to be able to receive translations. I can't merge because the tests are failing.

I can now do two things:
- remove the tests for the german locale and readd them after the merge with the translations (pretty inconvenient)
- manually edit the `de.json` file to add stubs for the translations

What would you recommend to do in a situation like this?

Reply:
Let's keep it as is then, no transifex. I will close this.

Reply:
I removed the project from transifex as well

Reply:
Thanks @emmenko 👍 Maybe we can revisit this in the future if we end up with a lot more translations...